### ISSUE TITLE
$(find) no longer resolved after other args

### ISSUE THREAD

**Issue by adamheins** [ros/xacro#338]
On the `noetic-devel` branch, resolving substitution arguments was outsourced to roslaunch. This resolves `$(find)` after other arguments like `$(arg)` (see [here](https://github.com/ros/ros_comm/blob/noetic-devel/tools/roslaunch/src/roslaunch/substitution_args.py#L380-L392)), which means I do things like pass `$(find)` into an argument. For example:
```xml
<xacro:arg name="robotfile" default="default.urdf.xacro"/>
<xacro:include filename="$(arg robotfile)"/>
```
and then I could pass in something like `robotfile:=$(find mypkg)/urdf/robot.urdf.xacro`, and expect the included filename to ultimately be resolved properly through the `$(arg)` and `$(find)`.

On the `ros2` branch, this logic has been changed so that the above no longer works for me. I'm wondering if this change was intended, or if I'm just doing something wrong here.

**Comment by rhaschke** [comment]
Fixed via #339. Do you need a release? Into which ROS2 distros?

**Comment by adamheins** [comment]
Thanks, that does it! I don't actually need a release; I came across this while working on [xacrodoc](https://github.com/adamheins/xacrodoc), which is a small tool that modifies xacro to (1) have more support for programmatically compiling files, and (2) makes the package-lookup mechanism fully functional even without ROS.
